service/resource/deployment/deployment.go

```diff
@@ -66,13 +66,23 @@ func (r Resource) newMainDeployment(cluster azuretpr.CustomObject) (Deployment,
 		"apiLoadBalancerCidr":           cluster.Spec.Azure.VirtualNetwork.LoadBalancer.APICIDR,
 		"etcdLoadBalancerCidr":          cluster.Spec.Azure.VirtualNetwork.LoadBalancer.EtcdCIDR,
 		"ingressLoadBalancerCidr":       cluster.Spec.Azure.VirtualNetwork.LoadBalancer.IngressCIDR,
+		"masterVMSize":                  cluster.Spec.Azure.Masters[0].VMSize,
+		"masterDataDiskSizeGB":          cluster.Spec.Azure.Masters[0].DataDiskSizeGB,
+		"masterAdminUsername":           cluster.Spec.Azure.Masters[0].AdminUsername,
+		"masterAdminSSHKeyData":         cluster.Spec.Azure.Masters[0].AdminSSHKeyData,
+		"masterOSImagePublisher":        cluster.Spec.Azure.Masters[0].OSImage.Publisher,
+		"masterOSImageOffer":            cluster.Spec.Azure.Masters[0].OSImage.Offer,
+		"masterOSImageSKU":              cluster.Spec.Azure.Masters[0].OSImage.SKU,
+		"masterOSImageVersion":          cluster.Spec.Azure.Masters[0].OSImage.Version,
```

Comment:
Are we going to support only a single master? I think we have to find a way to handle arrays in ARM templates.

Reply:
In the end, I think it would make more sens to update the [Masters and Workers fields](https://github.com/giantswarm/azuretpr/blob/master/spec/azure.go#L13) in the azuretpr to be of type azure.Node instead of []azure.Node.

In the end, I would assume that every masters will have the same specs, as well as the workers.

We would still be able to create multiple masters/workers by passing the [nodeCount parameter](https://github.com/giantswarm/azure-operator/pull/21/files#diff-bdfeb20a499ae8d7abc02b96a7ec37a5R262)

Reply:
For master nodes this is true. Not necessary for worker nodes. We have each node in the cluster configured separately in other operators, and it will make it easier to keep it that way.

@xh3b4sd you may want to provide more input here.

Reply:
+1 it should be possible to have workers with different capabilities running in your cluster. You may want to have a cluster with 6 nodes for workloads where 3 have a lot of computing power and 3 have a lot of memory or disk space.

Reply:
Makes sense to me, I'll start looking at how we could provide an arrow of masters/workers as parameters